fe/fe-core/src/main/java/com/starrocks/lake/Utils.java

```diff
@@ -113,6 +113,12 @@ public static void publishVersionBatch(@NotNull List<Tablet> tablets, List<TxnIn
         if (nodeToTablets == null) {
             nodeToTablets = new HashMap<>();
         }
+
+        // publish version operation should be successful even if the warehouse is not exist
+        if (!GlobalStateMgr.getCurrentState().getWarehouseMgr().warehouseExists(warehouseId)) {
+            warehouseId = WarehouseManager.DEFAULT_WAREHOUSE_ID;
```

Comment:
log a message to this situation. ideally this should not happen.

Also need to consider another situation where the warehouse exists but there is no compute node in the warehouse, what to do then?

Reply:
In this situation, FE will fail, and user can solve this problem by add compute node.

But If the warehouse is deleted, the new created warehouse with the same name will have a different warehouse ID.

Reply:
why bother user to add nodes into warehouse, can we move one step forward, choose nodes in other warehouse until there is no warehouse to choose to accomplish this task and then report failure to the end user?

Reply:
Have logged a message to this situation.

Reply:
Currently, we have assumed that nodes exist in the default warehouse. As for the logic of selecting other warehouses, I think it needs to be done after removing the reliance on the default warehouse. Currently, other parts of the code also use the default warehouse as a backup.

Reply:
Yes. so if a non-default warehouse doesn't have a node, it will fail there until user looks into the log and analysis the root cause, then get understand, the warehouse must be with at least one node in order to get this go through.

So if there is strong assumption, it will also fine to move to default warehouse, if 
* the given warehouse doesn't exist, **OR**
* the given warehouse DOES exist but DOESNOT have any node.

Reply:
The reason we let user add nodes into warehouse is that I think this is user's wrong operation, they can and should fix it themselves. But if the user delete the warehouse they can not fix it by themselves.

Reply:
what if there is not possible to fix it in near term. Failing the ops in the beginning if possible but not in the last step. this publish txn can't be cancelled and will block all the subsequent operations in the queue.

Reply:
It is purely valid to have a warehouse but with no node. There is no such restriction that a warehouse must have nodes. If the situation can be solved with less assumption and it is reasonable to remove the assumption, then we should go without the assumption.

Reply:
> what if there is not possible to fix it in short time. Failing the ops in the beginning if possible but not in the last step. this publish txn can't be cancelled and will block all the subsequent operations in the queue.

For public version operations, if no Node exists, it is reasonable to select the Node in the Default Warehouse.

Reply:
> It is purely valid to have a warehouse but with no node. There is no such restriction that a warehouse must have nodes. If the situation can be solved with less assumption and it is reasonable to remove the assumption, then we should go without the assumption.

I think we can totally fix this in another PR.